### ISSUE TITLE
OpenStack wrong tenants URL

### ISSUE THREAD

**Issue by aashah** [fog/fog#3662]
When calling `#tenants` on an OpenStack Fog::Compute connection, we get the following:

```
Expected([200, 204]) <=> Actual(300 Multiple Choices) excon.error.response :body => "{\"choices\": [{\"status\": \"CURRENT\", \"media-types\": [{\"base\": \"application/xml\", \"type\": \"application/vnd.openstack.compute+xml;version=2\"}, {\"base\": \"application/json\", \"type\": \"application/vnd.openstack.compute+json;version=2\"}], \"id\": \"v2.0\", \"links\": [{\"href\": \"http://REDACTED/v2/v2.0/tenants\", \"rel\": \"self\"}]}]}" :headers => { "Connection" => "close" "Content-Length" => "323" "Content-Type" => "application/json" "Date" => "Fri, 14 Aug 2015 17:05:31 GMT" } :local_address => "REDACTED" :local_port => 49873 :reason_phrase => "Multiple Choices" :remote_ip => "REDACTED" :status => 300 :status_line => "HTTP/1.1 300 Multiple Choices\r\n"
```

The problem may be around the `"http://REDACTED/v2/v2.0/tenants"` url, which has /v2/v2.0.

**Comment by domcleal** [comment]
We hit the same issue in Foreman with Fog 1.32.0 and started passing :openstack_identity_endpoint in with the same auth URL to fix it: https://github.com/theforeman/foreman/pull/2529/files

In Fog 1.33.0 it broke again, but passing :openstack_identity_public_endpoint instead seems to work.

**Comment by domcleal** [comment]
> In Fog 1.33.0 it broke again, but passing :openstack_identity_public_endpoint instead seems to work.

Regarding this, it appears that https://github.com/fog/fog/commit/6ecc87496b0f3bdb71fbd7c90cb8e4945dadd432#diff-3aa267766daa7f21447cd3c77b43bf39L74 broke this option in 1.33.0, since passing in `:openstack_identity_endpoint` used to result in `@openstack_identity_public_endpoint` being set.

Since that commit, it sets `@openstack_identity_endpoint` instead which isn't used in the code (e.g. https://github.com/fog/fog/commit/6ecc87496b0f3bdb71fbd7c90cb8e4945dadd432#diff-3aa267766daa7f21447cd3c77b43bf39L129).  The `:openstack_identity_endpoint` option is still documented as being the correct one here: https://github.com/fog/fog/blob/v1.33.0/lib/fog/openstack/compute.rb#L10 etc.

**Comment by geemus** [comment]
@TerryHowe @Ladas hey, sounds like it might not be too crazy to fix (and that it regressed). Could one of you take a look? Thanks!

**Comment by TerryHowe** [comment]
Did @dhague see this?

**Comment by Ladas** [comment]
seems like inconsistency of naming were there before
@openstack_identity_public_endpoint = options[:openstack_identity_endpoint]

hopefully @dhague can fix it, otherwise I can try. :-) Though I haven't hit this myself

**Comment by dhague** [comment]
I will investigate, but likely not until next week - I have the honour of being my team's on-duty support person this week, which tends to absorb all available capacity.

**Comment by dhague** [comment]
https://github.com/fog/fog/pull/3675 should fix this

**Comment by geemus** [comment]
Should be fixed by #3675

**Comment by domcleal** [comment]
Thanks!
